src/coreclr/jit/fgbasic.cpp

```diff
@@ -684,24 +682,41 @@ void Compiler::fgReplaceJumpTarget(BasicBlock* block, BasicBlock* oldTarget, Bas
                 {
                     assert(block->KindIs(BBJ_COND));
                     block->SetTrueEdge(newEdge);
-
-                    if (oldEdge->hasLikelihood())
-                    {
-                        newEdge->setLikelihood(oldEdge->getLikelihood());
-                    }
                 }
             }
             else
             {
                 assert(block->FalseTargetIs(oldTarget));
                 FlowEdge* const oldEdge = block->GetFalseEdge();
 
+                // Already degenerate cases should have taken the true path above.
+                assert(!block->TrueEdgeIs(oldEdge));
+
                 // fgRemoveRefPred should have removed the flow edge
                 fgRemoveRefPred(oldEdge);
                 assert(oldEdge->getDupCount() == 0);
                 FlowEdge* const newEdge = fgAddRefPred(newTarget, block, oldEdge);
-                block->SetFalseEdge(newEdge);
+
+                if (block->KindIs(BBJ_ALWAYS))
+                {
+                    block->SetTargetEdge(newEdge);
+                }
+                else
+                {
+                    assert(block->KindIs(BBJ_COND));
+                    block->SetFalseEdge(newEdge);
+                }
             }
+
+            if (block->KindIs(BBJ_COND) && (block->GetFalseEdge() == block->GetTrueEdge()))
```

Comment:
I noticed that we tend to compare the true/false edges with `TrueEdgeIs`. I'm planning on modifying `fgReplaceJumpTarget` in an upcoming PR, so I'll fix this then.